### ISSUE TITLE
Ctrl+Z to Exit the CLI Application

### ISSUE THREAD

**Issue by onebeartoe** [camrein/EzGraver#45]
Hi.  I am using the EzGraverCli application with a NEJE DK-8-KZ 1000mW, on Ubuntu.

The CLI application engraved the "Works!" image when I gave it the 's' argument.

The engraving came out nice!

But the application did not run to completion; I had to issue ctrl+z for it to end.

Is this a known issue on Linux/Ubuntu?

**Comment by onebeartoe** [comment]
Also is there any way to get feedback from the command line (the app's stdout I guess) on the progress of the current command?

**Comment by onebeartoe** [comment]
If there is anything I can do to help troubleshoot the ctrl+z issue, then I don't mind trying.

**Comment by camrein** [comment]
I'm sorry for the late reply as I was on vacation.
Thanks a lot for your report. As I only had a Mint VM by hand, I played around with the CLI version on it. I was able to reproduce your reported issue that the application does not run to completion. Moreover, I have noticed that I was unable to upload an image to the device. I suppose this is related to the situation that the different operating systems handle serial devices differently as well as the send and receive buffers. Furthermore, I think the issue was related to the asynchronous nature of QT (at least a combination thereof).

Nonetheless, I managed to fix the issue in the branch [unix_cli_fixes](https://github.com/camrein/EzGraver/tree/unix_cli_fixes). At least it is working now for both, Windows and Mint. I'd be glad if you could give it a shot and report back your experiences.

Cheers
Christoph

**Comment by onebeartoe** [comment]
Hey no problem, man.  Thanks for putting out the branch!  I am pulling it down now...

**Comment by onebeartoe** [comment]
Yes, sir!  It works with r and s command on the same Ubuntu machine.

Thanks!

**Comment by onebeartoe** [comment]
Yeah so I sent an image to the engraver and it came out as expected.  :)

So this issue ready to close (by my understanding).

On a side note though, is there a Wiki for this project?  I'd love to see how to setup an IDE for developement and maybe a walkthrough of the GUI app's features.

**Comment by camrein** [comment]
Happy to hear that. Thanks for the confirmation that everything's fine by now.

Unfortunately not. There is only the README.md itself. Qt Creator is the IDE I use for developing EzGraver. The usage is straightforward on Windows, just open the Qt project file (EzGraver.pro). I guess there is no significant difference for other operating systems.

**Comment by onebeartoe** [comment]
Yep, no problem. 

And thanks for the heads up on the 'clean' branch.
